### DISCUSSION TITLE
Currently Visible Space Names

### DISCUSSION THREAD

**Discussion by badassmexican** [asmvik/yabai#1564] [Q&A] [answered]
I'm trying to understand Jesse Skelton's switchAndFocusSIP.sh script to see if I can get it to work.  

I'm currently focusing on this line:

CurrentlyVisibleSpaceNames=$(yabai -m query --spaces | jq -re ".[] | select(.visible == 1)" | jq -re ".label")

If I run it in the command line it returns nothing.  

If I run just this part I see that there's and .is-visible property:

`yabai -m query --spaces | jq -re ".[]"
{
  "id": 3,
  "uuid": "EBC33C3D-9F5C-4C6A-A978-CF90567783A2",
  "index": 1,
  "label": "one",
  "type": "bsp",
  "display": 1,
  "windows": [
    7289,
    7288,
    40,
    3239,
    4077,
    4711
  ],
  "first-window": 7288,
  "last-window": 7289,
  "has-focus": true,
  "is-visible": true,
  "is-native-fullscreen": false
}`

How do I modify this line to return the labels of visible spaces?

yabai -m query --spaces | jq -re ".[] | select(.visible == 1)" | jq -re ".label"

I've tired changing .visible to .is-visible but I get an error.

**Comment by asmvik** [comment, marked as answer]
The following should do what you want: `yabai -m query --spaces | jq -re '.[] | select(."is-visible" == true)' | jq -re '.label'`
